### DISCUSSION TITLE
Support more LLMs

### DISCUSSION THREAD

**Discussion by Thore1954** [Christopher-Hayes/vscode-chatgpt-reborn#49] [Ideas]
Is there a plan to support LLMs other than those from OpenAI? There is [huggingface-vscode](https://github.com/huggingface/huggingface-vscode) that works with HF models but it doesn't have a chat interface.
Does anyone know of a similar plugin that works with LLMs on HF?

**Comment by Christopher-Hayes** [comment]
Right now, it's models that mimic OpenAI's API. I believe there are some GitHub projects that "proxy" other models to use OpenAI's API, which would allow this extension to use other models. Long-term it would be great to support all kinds of models, I don't see that happening in the near future, because that kind of refactoring will take time. But, we might start using Vercel's new [ai](https://sdk.vercel.ai/docs) library soon, which has "First-class support for [LangChain](https://github.com/vercel-labs/ai/blob/HEAD/js.langchain.com/docs) and [OpenAI](https://openai.com/), [Anthropic](https://www.anthropic.com/), [Cohere](https://cohere.com/) and [Hugging Face](https://huggingface.co/)".

**Reply by Thore1954** [reply]
So basically I can wrap HF's Inference Endpoints in a server that mimics OpenAI's API and set it in the `chatgpt.gpt3.apiBaseUrl` config? Didn't know that. Thanks.

**Reply by Christopher-Hayes** [reply]
Yeah, that's the idea. I think some people are doing that now for Claude. That config is being used a lot by people who geographically are blocked from accessing api.openai.com and instead use a proxy server. I haven't personally done anything with other models, so I can't offer much more detail than that.
